src/components/input/template.test.js

```diff
@@ -0,0 +1,105 @@
+/* globals describe, it, expect */
+
+const { render, htmlWithClassName } = require('../../../lib/jest-helpers')
+
+describe('Input', () => {
+  describe('by default', () => {
+    it('renders with classes', () => {
+      const $ = render('input', {
+        classes: 'app-c-input--custom-modifier'
+      })
+
+      const $component = $('.govuk-c-input')
+      expect($component.hasClass('app-c-input--custom-modifier')).toBeTruthy()
+    })
+
+    it('renders with id', () => {
+      const $ = render('input', {
+        id: 'my-input'
+      })
+
+      const $component = $('.govuk-c-input')
+      expect($component.attr('id')).toEqual('my-input')
+    })
+
+    it('renders with name', () => {
+      const $ = render('input', {
+        name: 'my-input-name'
+      })
+
+      const $component = $('.govuk-c-input')
+      expect($component.attr('name')).toEqual('my-input-name')
+    })
+
+    it('renders with type="text"', () => {
+      const $ = render('input')
+
+      const $component = $('.govuk-c-input')
+      expect($component.attr('type')).toEqual('text')
+    })
+
+    it('renders with value', () => {
+      const $ = render('input', {
+        value: 'QQ 12 34 56 C'
+      })
+
+      const $component = $('.govuk-c-input')
+      expect($component.val()).toEqual('QQ 12 34 56 C')
+    })
+
+    it('renders with attributes', () => {
+      const $ = render('input', {
+        attributes: {
+          'data-attribute': 'my data value'
+        }
+      })
+
+      const $component = $('.govuk-c-input')
+      expect($component.attr('data-attribute')).toEqual('my data value')
+    })
+
+    it('renders with label', () => {
+      const $ = render('input', {
+        id: 'my-input',
+        label: {
+          'text': 'National Insurance number'
+        }
+      })
+
+      expect(htmlWithClassName($, '.govuk-c-label')).toMatchSnapshot()
```

Comment:
Could we wrap these snapshots tests with a description 'dependant components' or similar?